### PR TITLE
Feature/seperated security schema files

### PR DESCRIPTION
**Pull request by jonaslagoni**
Fixes https://github.com/asyncapi/generator/issues/138

I was a bit split in regards to the naming of the functions and regards to the readme, let me know if anything should be changed or added :)

### CODE REVIEW

**Comment by jonaslagoni** [comment]
Question is, if we want to refactor the code to be able to split out any kind of component definition into separate files? -- https://www.asyncapi.com/docs/specifications/2.0.0/#componentsObject
Might make it easier to maintain

**Comment by jonaslagoni** [comment]
Seems like i did not test it fully through, gonna test it a bit further.

**Comment by jonaslagoni** [comment]
Regarding the refactoring, it could be done like the following (quick n dirty):
```
  generateSeperateFiles(asyncapiDocument, array, template, fileName, baseDir) {
    const promises = [];

    Object.keys(array).forEach((name) => {
      let component = array[name];
      promises.push(this.generateSeperateFile(asyncapiDocument, name, component, template, fileName, baseDir));
    });

    return Promise.all(promises);
  }
  async generateSeperateFile(asyncapiDocument, name, component, template, fileName, baseDir) {
    try {
      const relativeBaseDir = path.relative(this.templateDir, baseDir);
      const newFileName = fileName.replace(`\$\$${template}\$\$`, _.kebabCase(name));
      const targetFile = path.resolve(this.targetDir, relativeBaseDir, newFileName);
      const relativeTargetFile = path.relative(this.targetDir, targetFile);

      const shouldOverwriteFile = this.shouldOverwriteFile(relativeTargetFile);
      if (!shouldOverwriteFile) return;
      let temp = {
        name
      };
      temp[template] = component;
      const content = await this.renderFile(asyncapiDocument, path.resolve(baseDir, fileName), temp);

      await writeFile(targetFile, content, 'utf8');
    } catch (e) {
      throw e;
    }
  }
```
where we could remove all the other functions, and use it like the following:
```
else if (stats.name.includes('$$securityScheme$$')) {
            // this file should be handled for each in asyncapi.components.securityScheme
            await this.generateSeperateFiles(asyncapiDocument, asyncapiDocument.components().securitySchemes(), 'securityScheme', stats.name, root);
            const templateFilePath = path.relative(this.templateDir, path.resolve(root, stats.name));
            fs.unlink(path.resolve(this.targetDir, templateFilePath), next);
          }
```

File: lib/generator.js [hunk 1]
```diff
@@ -396,6 +401,55 @@ class Generator {
...
+      const relativeBaseDir = path.relative(this.templateDir, baseDir);
+      const newFileName = fileName.replace('$$securityScheme$$', _.kebabCase(securitySchemeName));
+      const targetFile = path.resolve(this.targetDir, relativeBaseDir, newFileName);
+      const relativeTargetFile = path.relative(this.targetDir, targetFile);
+
+      const shouldOverwriteFile = this.shouldOverwriteFile(relativeTargetFile);
+      if (!shouldOverwriteFile) return;
+      const content = await this.renderFile(asyncapiDocument, path.resolve(baseDir, fileName), {
+        securitySchemeName,
+        securityScheme: asyncapiDocument.components().securitySchemes()[securitySchemeName],
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply lib/generator.js]: I used with the 's' `securitySchemes`, which was why it did not work :)

**Comment by jonaslagoni** [comment]
You welcome, yes, will do another PR with the refactoring.

**Comment by fmvilas** [comment]
It's now on npm. Thanks!

**Pull request merged**